executor/executor_distsql.go

```diff
@@ -539,8 +539,10 @@ func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
 	selIdxReq := new(tipb.SelectRequest)
 	selIdxReq.StartTs = e.startTS
 	selIdxReq.IndexInfo = distsql.IndexToProto(e.table.Meta(), e.indexPlan.Index)
-	if e.indexPlan.Desc {
-		selIdxReq.OrderBy = append(selIdxReq.OrderBy, &tipb.ByItem{Desc: e.indexPlan.Desc})
+	if len(e.indexPlan.SortItems) > 0 {
+		selIdxReq.OrderBy = e.indexPlan.SortItems
+	} else if e.indexPlan.Desc {
```

Comment:
If e.indexPlan.SortItems > 0, we don't care if the table is desc scanned or not.